### PR TITLE
Icon improvements (fixes missing icons on recent core)

### PR DESCRIPTION
**Pull request by timja**
- Fix missing icons from recent core versions (swap to using supported icon class rather than unsupported URL)
- Use SVGs everywhere rather than different sized PNGs
- Use icon class spec throughout rather than referring to urls

Before:

image

After:

image

(p.s. there's a new API called only available since 2.340 which is far simpler, some docs at https://github.com/jenkinsci/design-library-plugin/blob/master/src/main/resources/io/jenkins/plugins/designlibrary/Symbols/index.jelly)

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/opentelemetry/OpenTelemetryRootAction.java [hunk 1]
```diff
@@ -22,25 +22,30 @@
     private OpenTelemetrySdkProvider openTelemetrySdkProvider;
 
     public Optional<ObservabilityBackend> getFirstMetricsCapableObservabilityBackend() {
-        final Optional<ObservabilityBackend> observabilityBackend = Optional.ofNullable(pluginConfiguration.getObservabilityBackends().stream().filter(backend -> backend.getMetricsVisualizationUrlTemplate() != null).findFirst().orElse(null));
+        final Optional<ObservabilityBackend> observabilityBackend = pluginConfiguration.getObservabilityBackends()
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/opentelemetry/OpenTelemetryRootAction.java:25]: this file is unrelated cleanup, can back out if desired, was just in here debugging why a link wasn't working after updating the icons

File: src/main/resources/io/jenkins/plugins/opentelemetry/job/MonitoringAction/jobMain.jelly [hunk 2]
```diff
@@ -2,12 +2,18 @@
...
     <table style="margin-top: 1em; margin-left:1em;">
-        <t:summary icon="/plugin/opentelemetry/images/48x48/opentelemetry.png">
-            OpenTelemetry:
-            <j:forEach items="${it.links}" var="link" varStatus="loop">
-                <img src="${resURL}${link.iconUrl}" height="24px"/><a href="${link.url}" target="_blank">${link.label}</a>
-                <j:if test="${!loop.last}">, </j:if>
-            </j:forEach>
+        <t:summary icon="icon-otel icon-xlg">
+            OpenTelemetry
+            <ul class="jenkins-!-padding-left-0">
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/opentelemetry/job/MonitoringAction/jobMain.jelly:7]: this class will only work on somewhat recent versions of core, but it looks okay without this just improves it

File: src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java [hunk 3]
```diff
@@ -531,7 +558,7 @@ private LogStorageRetriever resolveLogStorageRetriever() {
                     new GStringTemplateEngine().createTemplate("https://plugins.jenkins.io/opentelemetry/"), // TODO better documentation URL
                     TemplateBindingsProvider.of(
                         ObservabilityBackend.TemplateBindings.BACKEND_NAME, "See documentation on missing logs visualization URL",
-                        ObservabilityBackend.TemplateBindings.BACKEND_24_24_ICON_URL, "/plugin/opentelemetry/images/24x24/opentelemetry.png"));
+                        ObservabilityBackend.TemplateBindings.BACKEND_24_24_ICON_URL, "/plugin/opentelemetry/svgs/opentelemetry.svg"));
```
Comment (timja) [inline src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:561]: haven't tested this, not sure where it comes up atm

**Comment by timja** [comment]
I will double check tomorrow and get back to you.

I split this down from a larger branch and may have missed something

**Comment by timja** [comment]
I was testing on a weekly version which has a simpler jelly API.

This works now on older LTS and weekly:
image

image

**Review by cyrille-leclerc — APPROVED** [review]
